components/Comment/hooks/useSelection.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import XRange from "../lib/xrange/XRange";
+
+const useSelection = ({
+  ref,
+}): {
+  selectionXRange: any;
+  initialSelectionPosition: any;
+  resetSelectedPos: Function;
+} => {
+  const [selectionXRange, setSelectionXRange] = useState<null | any>(null);
+  const [initialSelectionPosition, setInitialSelectionPosition] = useState<
+    null | any
+  >(null);
+
+  const resetSelectedPos = () => {
+    setSelectionXRange(null);
+    setInitialSelectionPosition(null);
+  };
+
+  useEffect(() => {
```

Comment:
Is settimtout used to ensure that the selection "event" is kicked off to async queue so that click event happens before selection?

Reply:
Right, I put in the comment: Without the setTimeout, the selection object will not accurately hold a range object. The setTimeout of 0 ensure that the click event finishes first so that the range object is captured correctly. It's not the prettiest of solution but without it, we need to orchestrate `mousedown` and `mouseup` events